### PR TITLE
[do not merge] attempt at #704

### PR DESCRIPTION
**Pull request by Arnaudv6**
This is my first lambda function: hope I'm not messing anything.
I spotted an issue: maximized window sometimes receives the second button press.
Launch firefox with multiple (unloaded) tabs, double-click the title-bar.
Effect: firefox is maximized (cool) but the tab most close to cursor sometimes gets loaded...

### CODE REVIEW

**Comment by soreau** [comment]
For what it's worth, there is a 'Convert to draft' button toward the top right you can use to disallow merging.

**Comment by Arnaudv6** [comment]
Thanks @soreau. Though I just don't see it. I'll keep looking.

**Comment by ammen99** [comment]
> Thanks @soreau. Though I just don't see it. I'll keep looking.

It is located on the right...

**Comment by Arnaudv6** [comment]
Spotted it, with ^F. Thanks

**Comment by Arnaudv6** [comment]
OK, I have one question left:
do you wish a new option input/double_click_timeout, or do I leave hard-coded value of 300ms?

**Comment by ammen99** [comment]
> OK, I have one question left:
> do you wish a new option input/double_click_timeout, or do I leave hard-coded value of 300ms?

If anything, the option should be in the decoration plugin because only it is affected by it.

However, I don't think an option here is required, unless someone requests it.

File: plugins/decor/deco-layout.cpp [hunk 1]
```diff
@@ -230,7 +231,20 @@ decoration_layout_t::action_response_t decoration_layout_t::handle_press_event(
...
         {
-            return {DECORATION_ACTION_MOVE, 0};
+            if (timer.is_connected()) /* we have a double-click (before release) */
+            {
+                return {DECORATION_ACTION_TOGGLE_MAXIMIZE, 0};
+            } else
+            {
+                timer.set_timeout(300, [=] ()
+                {
+                    timer.disconnect();
```
Comment (ammen99) [inline plugins/decor/deco-layout.cpp:241]: you don't need a disconnect here, `return false` is enough.

File: plugins/decor/deco-layout.cpp [hunk 2]
```diff
@@ -230,7 +231,20 @@ decoration_layout_t::action_response_t decoration_layout_t::handle_press_event(
...
+                    return false;
+                });
+                // hardcoded 300ms, to be replaced by option in config:
+                // wf::option_wrapper_t<int> delay{"input/double_click_timeout"};
+                return {DECORATION_ACTION_MOVE, 0};
```
Comment (ammen99) [inline plugins/decor/deco-layout.cpp:246]: I don't think this is very nice, we start a move action (potentially triggering wobbly & co) only to immediately exit it. Can we instead modify the code a bit more: trigger move on pointer motion if we have clicked on an area suitable for moving. (you'd also need to change the return type of the `handle_motion` method and handle it in deco-subsurface.cpp)

**Review by ammen99 — CHANGES_REQUESTED** [review]

File: plugins/decor/deco-layout.cpp (hunk 2 above)
Comment (Arnaudv6) [reply plugins/decor/deco-layout.cpp:244]: agreed. (did not see as I don't use wobbly).
I just filled #1074 which is related but more complicated.

**Review by Arnaudv6 — COMMENTED** [review]
closing this and opening another pull request to modify deco-subsurface.cpp also.

**Pull request closed without merging**